packages/components/forms/src/text-input/TextInput.tsx

```diff
@@ -22,43 +23,42 @@ export const _TextInput = (
   }: TextInputProps,
   ref: React.Ref<HTMLInputElement>,
 ) => {
-  const [valueState, setValueState] = useState(value);
+  const textInputRef = useRef(null);
   const styles = getStyles();
 
-  // Store a copy of the value in state.
-  // This is used by this component when the `countCharacters`
-  // option is on
-  const handleOnChange = useCallback(
-    (e: ChangeEvent<HTMLInputElement>) => {
-      setValueState(e.target.value);
-      if (onChange) onChange(e);
-    },
-    [onChange],
-  );
-
   const copyButtonStyles = cx(styles.copyButton, {
     [styles.disabled]: Boolean(isDisabled),
     [styles.invalid]: Boolean(isInvalid),
   });
 
+  const handleCopy = (e) => {
+    if (onCopy) {
+      onCopy(e);
+    }
+
+    textInputRef.current.select();
+    document.execCommand('copy');
```

Comment:
does it work in all browsers we support?

Reply:
yeah.